### ISSUE TITLE
toSparql for MINUS throws error

### ISSUE THREAD

**Issue by rubensworks** [joachimvh/SPARQLAlgebra.js#75]
When parsing the following query, and generating a SPARQL query from it again throws an error in SPARQL.js

```sparql
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>

SELECT *
WHERE {
    ?uri skos:prefLabel ?prefLabel
    MINUS {
        ?uri skos:broader <https://data.cultureelerfgoed.nl/term/id/cht/522f659d-6dc5-4b64-a244-1ea88701b7ea>
    }
}

```

```
TypeError: array.map is not a function
    at mapJoin (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:430:16)
    at Generator.array (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:103:10)
    at Generator.group (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:153:34)
    at Generator.minus (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:179:26)
    at Generator.toPattern (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:95:20)
    at Array.map (<anonymous>)
    at mapJoin (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:430:16)
    at Generator.array (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:103:10)
    at Generator.toPattern (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:95:20)
    at Generator.group (/Users/rtaelman/Documents/UGent/nosync/SPARQL.js/lib/SparqlGenerator.js:154:34)
```

I however suspect that SPARQL.js is not at fault here, but simply receives an invalid input to `SparqlGenerator`.
I've placed some debug logs in SPARQLAlgebrajs's `toSparqlJs`, and it looks like `translateOperation` produces the following:

```json
{
  "type": "query",
  "prefixes": {},
  "queryType": "SELECT",
  "variables": [
    {
      "termType": "Variable",
      "value": "uri"
    },
    {
      "termType": "Variable",
      "value": "prefLabel"
    }
  ],
  "where": [
    {
      "type": "query",
      "prefixes": {},
      "queryType": "SELECT",
      "variables": [
        {
          "termType": "Variable",
          "value": "uri"
        },
        {
          "termType": "Variable",
          "value": "prefLabel"
        }
      ],
      "where": [
        {
          "type": "bgp",
          "triples": [
            {
              "subject": {
                "termType": "Variable",
                "value": "uri"
              },
              "predicate": {
                "termType": "NamedNode",
                "value": "http://www.w3.org/2004/02/skos/core#prefLabel"
              },
              "object": {
                "termType": "Variable",
                "value": "prefLabel"
              }
            }
          ]
        },
        {
          "type": "minus",
          "patterns": {
            "type": "bgp",
            "triples": [
              {
                "subject": {
                  "termType": "Variable",
                  "value": "uri"
                },
                "predicate": {
                  "termType": "NamedNode",
                  "value": "http://www.w3.org/2004/02/skos/core#broader"
                },
                "object": {
                  "termType": "NamedNode",
                  "value": "https://data.cultureelerfgoed.nl/term/id/cht/522f659d-6dc5-4b64-a244-1ea88701b7ea"
                }
              }
            ]
          }
        }
      ]
    }
  ]
}
```

It looks like SPARQL.js expects an array for the `patterns` field for MINUS, while SPARQLAlgebra.js provides it as a simple hash. I don't think that the fix will be as simple as wrapping the hash in an array, as it probably needs to be combined with the BGP above (which seems to be disconnected from the MINUS clause, which seems incorrect as well).

Originally reported in https://github.com/comunica/comunica/issues/779

**Comment by joachimvh** [comment]
Will probably be able to look into this next week. Unfortunately the `toSparql` part is less robust then the algebra part.

**Comment by joachimvh** [comment]
It actually was as simple as wrapping the hash in an array. This is how SPARQL.js handles MINUS operations: https://github.com/RubenVerborgh/SPARQL.js/blob/main/test/parsedQueries/sparql/sparql-8-3-1b.json

Fixed in eb71a9291ac341eaf162e002515d96b2aa6b4ef1

**Comment by rubensworks** [comment]
Ah ok, then I was making it more difficult than needed 😄 
Thanks for the fix!
